apps/notification-producer/src/utils/getNotificationSummary.ts

```diff
@@ -0,0 +1,38 @@
+import { ALL_SUPPORTED_CHAINS_MAP, SupportedChainId } from '@cowprotocol/cow-sdk';
+import { getAddress } from 'viem';
+import { ChainNames, formatAmount, formatTokenName } from '@cowprotocol/shared';
+import { Erc20Repository } from '@cowprotocol/repositories';
+
+interface OrderInfoForNotificationParams {
+  chainId: SupportedChainId;
+  isEthFlowOrder: boolean;
+  erc20Repository: Erc20Repository;
+  sellTokenAddress: string;
+  buyTokenAddress: string;
+  sellAmount: string | bigint;
+  buyAmount: string | bigint;
+}
+
+export async function getNotificationSummary(params: OrderInfoForNotificationParams): Promise<string> {
+  const { chainId, isEthFlowOrder, erc20Repository } = params;
+
+  const sellToken = isEthFlowOrder
+    ? ALL_SUPPORTED_CHAINS_MAP[chainId].nativeCurrency
+    : await erc20Repository.get(
+      chainId,
+      getAddress(params.sellTokenAddress)
+    );
```

Comment:
I mentioned in the other PR, erc20Repository can handle native tokens.

```suggestion
  const sellToken = await erc20Repository.get(
      chainId,
      isEthFlowOrder? NATIVE_TOKEN_ADDRESS : getAddress(params.sellTokenAddress)
    );
```